### DISCUSSION TITLE
How do I refresh my AWS S3 credentials for logging artifacts?

### DISCUSSION THREAD

**Discussion by seanswyi** [mlflow/mlflow#10920] [Q&A]
I run training scripts that typically take _at least_ a day or two. It's not uncommon for me to find out that my program has crashed due to the provided token expiring.

I've looked into why this happens and it seems like normally when using Boto3's S3 client I could create a custom class for this (reference: https://stackoverflow.com/questions/63724485/how-to-refresh-the-boto3-credentials-when-python-script-is-running-indefinitely) but I'm not sure how to implement that using MLflow. I'm currently using MLflow with HuggingFace Accelerate's MlflowTracker object as follows:

```Python
import os

from accelerate.tracking import MLflowTracker, on_main_process


class CustomMLflowTracker(MLflowTracker):
    @on_main_process
    def __init__(
        self,
        mlflow_uri: str = None,
        experiment_name: str = None,
        run_id: Optional[str] = None,
        nested_run: Optional[bool] = False,
        run_name: Optional[str] = None,
        description: Optional[str] = None,
    ):
        experiment_name = os.getenv("MLFLOW_EXPERIMENT_NAME", experiment_name)
        run_id = os.getenv("MLFLOW_RUN_ID", run_id)
        nested_run = os.getenv("MLFLOW_NESTED_RUN", nested_run)

        mlflow.set_tracking_uri(mlflow_uri)

        experiment = mlflow.set_experiment(
            experiment_name=experiment_name,
        )
        experiment_id = experiment.experiment_id

        self.active_run = mlflow.start_run(
            run_id=run_id,
            experiment_id=experiment_id,
            run_name=run_name,
            nested=nested_run,
            description=description,
        )



tracker = CustomMLflowTracker(
    mlflow_uri=args.mlflow_uri,
    experiment_name=args.experiment,
    run_name=run_name,
    )
accelerator_kwargs["log_with"] = tracker


accelerator = Accelerator(
    gradient_accumulation_steps=args.gradient_accumulation_steps,
    mixed_precision=args.mixed_precision,
    **accelerator_kwargs,
)

experiment_config = vars(args)
accelerator.init_trackers(args.experiment, experiment_config)
```

And when I log my stuff to my AWS S3/MLflow, I just do `mlflow.log_artifact`. Right now I've tried to just put the artifact logging inside of a try-except and trying again but I'm not sure if that's the optimal solution.

How can I make sure that my credentials issue doesn't crash the program?

**Comment by jayanthreddy1997** [comment]
@seanswyi Were you able to find a fix for this issue?

**Reply by seanswyi** [reply]
No, I don't believe I did... I think that I just had to find a workaround online for it. It's been so long though, I don't really remember.
